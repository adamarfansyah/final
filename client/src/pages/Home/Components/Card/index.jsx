import PropTypes from 'prop-types';
import classes from './style.module.scss';

const Card = ({ sport }) => (
  <div className={classes.card}>
    <div className={classes.icons}>
      <img src={sport.icon} alt={sport.name} className={classes.icon} />
    </div>
    <div className={classes.name}>{sport.name}</div>
    <div>{sport.desc}</div>
  </div>
);

Card.propTypes = {
  sport: PropTypes.object,
};

export default Card;
