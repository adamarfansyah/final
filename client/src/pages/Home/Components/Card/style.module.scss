@import '@styles/base';

.card {
  flex: 1;
  background-color: var(--color-bg);
  padding: 2rem;
  border-radius: 2rem;
  box-shadow: 1rem 1rem 1rem rgba($color: #000000, $alpha: 0.2);
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
  .icons {
    border-radius: 100%;
    width: fit-content;
    background-color: white;
    padding: 1rem;
  }
  .icon {
    width: 3rem;
    color: white;
  }
  .name {
    font-size: 1.2rem;
    font-weight: bold;
  }
}
