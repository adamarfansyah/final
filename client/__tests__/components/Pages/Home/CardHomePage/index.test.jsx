import { render } from '@utils/testHelper';
import Card from '@pages/Home/Components/Card';
import badminton from '@static/images/badminton.svg';

const mockProps = {
  id: 1,
  name: 'Badminton',
  icon: badminton,
};

let wrapper;

beforeEach(() => {
  wrapper = render(<Card sport={mockProps} />);
});

describe('Card Home Page', () => {
  test('Should render correctly', () => {
    const { getByRole, getByText } = wrapper;
    const icon = getByRole('img', { name: /badminton/i });
    const text = getByText(/badminton/i);
    expect(text).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
  });

  test('Should match with snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});
