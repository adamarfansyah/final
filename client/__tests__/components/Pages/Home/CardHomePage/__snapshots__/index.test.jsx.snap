// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Home Page Should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="icons"
    >
      <img
        alt="Badminton"
        class="icon"
      />
    </div>
    <div
      class="name"
    >
      Badminton
    </div>
  </div>
</DocumentFragment>
`;
